reference/functions/pgroonga-query-expand.md

```diff
@@ -129,6 +129,71 @@ SELECT pgroonga_query_expand('synonym_groups', 'synonyms', 'synonyms',
 -- (1 row)
 ```
 
+### Practical Example Using Synonym groups
+
+Sometimes you may want to search similar names like (Timothy or Tim), (William or Bill), (Stephen, Steven or Steve).
+
+Here are sample schema and data for solving this problem using Synonym groups.
+
+#### Name Table
+
+```sql
+CREATE TABLE names (
+  name varchar(255)
+);
+
+CREATE INDEX pgroonga_names_index
+          ON names
+       USING pgroonga (name pgroonga_varchar_full_text_search_ops_v2);
+
+INSERT INTO names
+  (name)
+  VALUES ('William Gates'),('Steven Paul Jobs'),('Timothy Donald Cook');
+```
+
+#### Synonym groups Table
+
+```sql
+CREATE TABLE synonym_groups (
+  synonyms text[]
+);
+
+CREATE INDEX synonym_groups_synonyms
+          ON synonym_groups
+       USING pgroonga (synonyms pgroonga_text_array_term_search_ops_v2);
+
+INSERT INTO synonym_groups
+  VALUES (ARRAY['Stephen', 'Steven', 'Steve']);
+```
+
+When you search "Steve" in this example, all of `"Stephen"`, `"Steven"` and `"Steve"` in query are expanded because PGroonga index is used.
+
+```sql
+SELECT pgroonga_query_expand('synonym_groups', 'synonyms', 'synonyms',
+                             'Steve') AS query_expand;
+--              query_expand             
+-- --------------------------------------
+--   ((Stephen) OR (Steven) OR (Steve))
+-- (1 row)
+```
+
+
+An example down below is Name Table Search Example with pgroonga_query_expand.
+
+Note: Name Table "name" column is `varchar` character type, so that you need to specifically type cast `pgroonga_query_expand` as `pgroonga_query_expand(...)::varchar` (BTW return type of `pgroonga_query_expand()` is `text` charactor type. You do not need to type cast `pgroonga_query_expand()` as varchar when you search on `text` charactor columns.)
```

Comment:
```suggestion
Note: Name Table "name" column is `varchar` character type. so that you need to specify type such as `pgroonga_query_expand(...)::varchar` (However, you do not need to type cast `pgroonga_query_expand()` as varchar when you search on `text` character columns. Because return type of pgroonga_query_expand()` is `text` character type.)
```